Parsers/Parser_DORY_to_HW.py

```diff
@@ -131,25 +131,29 @@ def update_branches_graph(self):
                                     nodes_scan_2.add_existing_parameter("branch_last", 0)
                                     break
                                 else:
-                                    nodes_scan_2.add_existing_parameter("branch_change", 1)  
-                                    node.add_existing_parameter("branch_last", 1)   
+                                    nodes_scan_2.add_existing_parameter("branch_change", 1)
+                                    node.add_existing_parameter("branch_last", 1)
                                     break
                             else:
                                 if(i < j):
                                     node.add_existing_parameter("branch_last", 1)
                                 else:
-                                    nodes_scan_2.add_existing_parameter("branch_last", 1)  
+                                    nodes_scan_2.add_existing_parameter("branch_last", 1)
 
     def update_dimensions_graph(self):
         print("\nUpdating dimensions of vectors inside the graph, if they do not match among nodes")
         for i, node in enumerate(self.DORY_Graph):
             if i > 0:
-                if isinstance(self.DORY_Graph[i].input_channels, type(None)):
+                # Scheremo: Fix bug where unhandled exception is thrown whenever
+                # input_channels is not defined...
+                if hasattr(self.DORY_Graph[i], 'input_channels') and isinstance(self.DORY_Graph[i].input_channels, type(None)):

```

Comment:
I am not sure this is handling any case correctly - but imo if a graph-level attribute is expected but not available, an exception should be thrown from "the first parser".